docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+# Simperby: Protocol Overview
+
+Simperby is a blockchain engine that runs an organiziation which is
+
+1. **Decentralized**
+2. **Standalone, Sovereign and Self-hosted**
+3. **With a set of permissioned members**
+
+in that it provides
+
+1. **Consensus**: a safe and live mechanism to finalize the state of the organization, which is tolerant to byzantine faults in an asynchronous network.
+2. **Governance**: a democratic mechanism to make decisions on the organization, which can be finalized by the consensus.
+3. **Communication Channel**: a mechanism to communicate with each other in a decentralized, verifiable, and fault-tolerant way.
+
+## Basics and Terminologies
+
+The reader of this document should be familiar with the basic concepts of blockchain.
+
+- Simperby is an engine that can build an instance of a blockchain.
+  Any blockchain built with Simperby is sometimes called a *Simperby chain*.
+- The unit of state transition is called a *transaction*.
+  `StoreData`, `DelegateGovernanceVotingPower`, `AddValidator` are the examples.
+- A sequence of transactions is called an *agenda*.
+  **Any governance voting is performed on an agenda and the target block height**.
+- A *block* consists of the header and **a single agenda, which is mandatory**.
+- The participants of the governance are called *members*.
+- The participants of the consensus are called *validators*, which is a subset of the *members*.
+- A *consensus leader* or a *block proposer* is a validator that proposes a block for the round.
+- A *round* is a period of time in which a consensus leader proposes a block, and consensus votes(pre-vote, pre-commit) are performed. Refer to Tendermint for more details.
+- A validator is *honest* or *not byzantine* if they follow the consensus protocol.
+- A block proposer is *faithful* if they fulfill their responsibility and *good* if they don't overuse their power. They are *lazy* and *bad* otherwise.
+  The notion of *responsibility* and *power* will be explained [later](#consensus-leader).
+
+## Summary
+
+Here is a summarized version of the Simperby protocol.
+
+1. We want the node operation to be **simple and lightweight** to realize a truly distributed, decentralized and self-hosted organization.
+2. To be so, it is assumed that most of the nodes are **rarely online** and the protocol produces new blocks **on-demand**.
+3. To accomplish that, the **consensus round must be very long**.
+4. For every block, members can **vote on or propose an agenda, propagating** to each other.
+5. At the same time, members can also **propagate their chats** used for the communication, which is **coordinated by the block proposer**.
```

Comment:
I think the word "coordinate" is not that appropriate. How about using "gather" or "collect" instead?

Reply:
The proposer doesn't just collect the chats. It frequently semifinalize the chat chain, and that's the point.

Reply:
I see